lib/manager/npm/post-update/index.ts

```diff
@@ -157,12 +157,15 @@ export async function writeExistingFiles(
       }
     }
     if (packageFile.yarnrc) {
-      logger.debug(`Writing .yarnrc to ${basedir}`);
-      const yarnrcFilename = upath.join(basedir, '.yarnrc');
+      const {
+        fileName: yarnrcFilename,
+        content: yarnrcContent,
+      } = packageFile.yarnrc;
+      logger.debug(`Writing ${yarnrcFilename} to ${basedir}`);
       try {
         await outputFile(
           yarnrcFilename,
-          packageFile.yarnrc
+          yarnrcContent
             .replace('--install.pure-lockfile true', '')
             .replace('--install.frozen-lockfile true', '')
```

Comment:
Are you sure this works for the new yaml file? I think we need different config options for yarn v2

Reply:
Yes, it doesn't work. But if the flag doesn't exist in the file, then there is nothing to replace.